opal/mca/if/linux_ipv6/if_linux_ipv6.c

```diff
@@ -77,7 +77,21 @@ opal_if_base_component_t mca_if_linux_ipv6_component = {
     },
 };
 
-/* configure using getifaddrs(3) */
+static int hex2int(char hex)
+{
+    if ('0' <= hex && hex <= '9') return hex - '0';
+    if ('A' <= hex && hex <= 'F') return hex - 'A' + 10;
+    if ('a' <= hex && hex <= 'f') return hex - 'a' + 10;
+    abort();
```

Comment:
This is a bit anti-social -- if an unexpected misconfiguration occurs, this could end up dropping core on tens / hundreds / thousands of times across a cluster.  Could you make this a bit more friendly of a failure mode?  Perhaps a call to `opal_show_help()` followed by a call to `exit(1)`, or somesuch.  I realize this is an unlikely error, but I don't think we want to take the chance of dropping TB worth of corefiles on a networked filesystem, for example.

Reply:
I though that since this file is Linux specific and Linux is very stable in this regard, we can expect the `/proc/net/if_inet6` format to never change, and this function to always succeed.

Please see the second commit that makes it robust.

Reply:
You're probably not wrong, but the last time I used `abort()` in a situation that should never have reached that code, a few months later, something else happened in the source to make that condition actually occur -- which resulted in `abort()` being called in an embedded controller.  This filled up the embedded filesystem on dozens on machines, bricking them.  QA was *NOT* happy with me.  😦 

So I've always been super shy of using `abort()` ever since then.  😄